src/main/java/storage/WriteDataFiles.java

```diff
@@ -1,33 +1,44 @@
 package storage;
 
 import cheatsheet.CheatSheet;
+import cheatsheet.CheatSheetList;
 
 import java.io.IOException;
 import java.io.FileWriter;
 
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.ArrayList;
 
-import static storage.ParseCheatSheetFile.NAME;
-import static storage.ParseCheatSheetFile.PROGRAMMING_LANGUAGE;
-import static storage.ParseCheatSheetFile.DETAILS;
+import static storage.ParseDataFile.NAME;
+import static storage.ParseDataFile.PROGRAMMING_LANGUAGE;
+import static storage.ParseDataFile.DETAILS;
+import static storage.ParseDataFile.WHITESPACE;
 
 /**
  * Allows the user to write data based on the cheatSheets currently present
- * in the list of cheatSheets into individual .txt files.
+ * in the list of cheatSheets into individual files.
  */
-public class WriteDataFile extends DataFile {
-    private static final String FILE_EXTENSION = ".txt";
+public class WriteDataFiles extends DataFile {
+    private ArrayList<CheatSheet> cheatSheets;
+    private Path debugPath = null;
 
-    // WriteDataFile Constructor
-    public WriteDataFile() {
+    // Main Constructor
+    public WriteDataFiles() {
+        executeFunction();
+    }
+
+    // Constructor for debugging purposes
+    public WriteDataFiles(Path textFilePath) {
+        this.cheatSheets = CheatSheetList.getCheatSheetList();
+        this.debugPath = textFilePath;
         executeFunction();
     }
 
     /**
      * Converts the cheatSheets present in the list of cheatSheet into
-     * a string. This string is then saved in a form of .txt file, 1 file
+     * a string. This string is then saved in a file, 1 file
      * for each cheatSheet.
      */
     @Override
```

Comment:
Isn't it better to just write the stuffs in storeCheatSheet() inside executeFunction() directly?